### PR TITLE
Convertinging util.js to typescript 3/3

### PR DESCRIPTION
**Pull request by waylandli**
Problem
=======
Github Issue #28
Part 3 of 3
Converting util.js to typescript

Change summary:
---------------
* Converted util.js to typescript

Steps to Verify:
----------------
* Tests should still run while the file is in typescript

### CODE REVIEW

File: lib/util.ts [hunk 1]
```diff
@@ -1,16 +1,22 @@
...
+  readPos: number
+  constructor(inBuf: Buffer) {
+    super(inBuf)
+    this.inBuf = inBuf
+    this.readPos = 0
+  }
+  readString(len: number) {
     this.ensureAvailable(len);
     var buffer = this.inBuf.slice(this.readPos, this.readPos + len);
     var str = this.inBuf.toString('utf8', this.readPos, this.readPos + len);
```
Comment (waylandli) [inline lib/util.ts:22]: Tried to override the super class version of readString but its not being allowed with different parameters

File: lib/util.ts [hunk 2]
```diff
@@ -31,25 +37,25 @@ const PageLocation = parquet_thrift.PageLocation.prototype = [];
...
 Object.defineProperty(PageLocation,'offset', getterSetter(0));
 Object.defineProperty(PageLocation,'compressed_page_size', getterSetter(1));
 Object.defineProperty(PageLocation,'first_row_index', getterSetter(2));
 
 
-exports.force32 = function() {
+export const force32 = function() {
   const protocol = thrift.TCompactProtocol.prototype;
   protocol.zigzagToI64 = protocol.zigzagToI32;
   protocol.readVarint64 = protocol.readVarint32 = function() {
```
Comment (waylandli) [inline lib/util.ts:53]: There is no type definition for these files in the imported types module but it exists in the actual module

File: lib/util.ts [hunk 3]
```diff
@@ -163,31 +171,31 @@ exports.fclose = function(fd) {
...
       } else {
-        resolve();
+        resolve(err);
       }
     });
   });
 }
 
-exports.osend = function(os) {
+export const osend = function(os) {
```
Comment (waylandli) [inline lib/util.ts:186]: Not sure what the type for os should be

**Comment by waylandli** [comment]
A lot of `//@ts-ignore` used because a lot of imported classes don't contain typings for their functions which causes typescript errors.

File: lib/util.ts [hunk 4]
```diff
@@ -27,29 +35,34 @@ class fixedTFramedTransport extends thrift.TFramedTransport {
   */
 
 const previousPageLocation = parquet_thrift.PageLocation.prototype;
+//@ts-ignore
```
Comment (shannonwells) [inline lib/util.ts:38]: I haven't been following along on these conversions - are we just using `@ts-ignore` for now?  Is there a plan to convert these to proper TypeScript?  @wilwade

**Review by shannonwells — APPROVED** [review]
Approved assuming there's some plan for removing all the @ts-ignores.

**Review by wilwade — APPROVED** [review]
💯

**Pull request merged**